cmd/kubeseal/main.go

```diff
@@ -71,17 +71,17 @@ var (
 	reEncrypt      bool // re-encrypt command
 	unseal         = flag.Bool("recovery-unseal", false, "Decrypt a sealed secrets file obtained from stdin, using the private key passed with --recovery-private-key. Intended to be used in disaster recovery mode.")
 	privKeys       = flag.StringSlice("recovery-private-key", nil, "Private key filename used by the --recovery-unseal command. Multiple files accepted either via comma separated list or by repetition of the flag. Either PEM encoded private keys or a backup of a json/yaml encoded k8s sealed-secret controller secret (and v1.List) are accepted. ")
+	kubeconfig     = flag.String("kubeconfig", "", "Path to a kube config. Only required if out-of-cluster")
+
+	globalOverrides *clientcmd.ConfigOverrides // config overrides from flags
 
 	// VERSION set from Makefile
 	VERSION = buildinfo.DefaultVersion
-
-	clientConfig clientcmd.ClientConfig
-
-	// testing hook for clientConfig.Namespace()
-	namespaceFromClientConfig = func() (string, bool, error) { return clientConfig.Namespace() }
 )
 
-func init() {
+type namespaceFn func() (string, bool, error)
+
+func initFlags() {
```

Comment:
Why are we replacing `init` with `initFlags`?

Reply:
`init()` is magic, it is always executed before the `main` function.

It means that such initialization is happening no matter what, and in this case that is NOT required. You can make the case for when you are, for instance, loading device or SQL driver implementations coming from dependencies.

Using `initFlags` instead we have full control on when the flags are parsed or not.